File: in_Java/1128 Number of Equivalent Domino Pairs.java
class Solution {
    public int numEquivDominoPairs(int[][] dominoes) {
        HashMap<int[], Integer> hm = new HashMap<>();
        boolean f;
        for(int i = 0; i < dominoes.length; i++){
            f = false;
            int[] cur = dominoes[i];
            Arrays.sort(cur);
            for(int[] k : hm.keySet()){
                if(k[0] == cur[0] && k[1] == cur[1]){
                    hm.put(k, hm.get(k)+1);
                    f = true;
                }
            }
            if(!f) hm.put(cur,1);
        }
        int res = 0;
        for(int i : hm.values()){
            //System.out.println(i);
            if(i > 1){
                res += i * (i-1)/2;
            }
        }
        return res;
    }
}

File: in_Python/1128 Number of Equivalent Domino Pairs.py
class Solution(object):
    def numEquivDominoPairs(self, dominoes):
        """
        :type dominoes: List[List[int]]
        :rtype: int
        """
        d = {}
        for dm in dominoes:
            dm.sort()
            dm = str(dm[0]) + str(dm[1])
            if dm not in d:
                d[dm] = 1
            else:
                d[dm] += 1
        res = 0
        for k in d:
            num = d[k]
            res += num * (num-1) / 2
        return res